src/components/releases/Album.js

```diff
@@ -84,9 +84,15 @@ function Album({ album }) {
  * @param {{ album: Album, uriLinks: boolean }} props
  */
 function renderArtists({ album, uriLinks }) {
-  const primary = Object.values(album.artists)
-    .flat()
-    .map((artist) => renderArtist({ artist, uriLinks, className: 'Album__artist' }))
+  const primaryArtists = [...new Set(Object.values(album.artists).flat())]
+
+  const uniquePrimaryArtists = [
+    ...new Map(primaryArtists.map((artist) => [artist['id'], artist])).values(),
+  ]
+
+  const primary = uniquePrimaryArtists.map((artist) =>
+    renderArtist({ artist, uriLinks, className: 'Album__artist' })
+  )
```

Comment:
I appreciate the effort, but this removes duplicates at the render stage, while I would like to prevent duplicates from being saved to the state at all. Can I kindly ask you to rollback these changes? Thank you for finding this, I will figure it out.